Src/FluentAssertions/Specialized/DelegateAssertions.cs

```diff
@@ -113,12 +113,15 @@ public AndConstraint<TAssertions> NotThrow<TException>(string because = "", para
 
             Type expectedType = typeof(TException);
 
-            Execute.Assertion
+            success = Execute.Assertion
                 .ForCondition(exception is not null)
                 .BecauseOf(because, becauseArgs)
                 .FailWith("Expected {0}{reason}, but no exception was thrown.", expectedType);
 
-            exception.Should().BeOfType(expectedType, because, becauseArgs);
+            if (success)
```

Comment:
We seem to have the same bug in `ThrowExactly<TException>`.
Wanna fix that as well?

Reply:
Sure :)